src/lib.rs

```diff
@@ -97,6 +97,9 @@ pub enum Error {
     Policy(policy::Error),
     /// Program does not have maximal sharing
     SharingNotMaximal,
+    /// Finalization led to the root node being hidden, because not enough witnesses
+    /// were provided (or some branches incomplete branches were kept instead of pruned).
+    IncompleteFinalization,
```

Comment:
86c4195171d58817d5da5aa60a1db9da93267e07: This error variant is not being used and it is confusing. Why would we ever hide the root node?